src/plots/cartesian/axis_autotype.js

```diff
@@ -81,3 +84,10 @@ function category(a) {
 
     return curvecats > curvenums * 2;
 }
+
+// very-loose requirements for multicategory,
+// trace modules that should never auto-type to multicategory
+// should be declared with 'noMultiCategory'
+function multiCategory(a) {
+    return Array.isArray(a[0]) && Array.isArray(a[1]);
```

Comment:
We decided these should be `isArrayOrTypedArray`, didn't we?

Reply:
Good eye -> https://github.com/plotly/plotly.js/pull/3300/commits/a9eca0951f568d4202dcb0fff8f5546948c27889